contentcuration/contentcuration/node_metadata/annotations.py

```diff
@@ -35,6 +36,49 @@ def build_topic_condition(self, kind_id, comparison='='):
         return self.build_kind_condition(kind_id, content_kinds.TOPIC, comparison)
 
 
+class AncestorAnnotation(MetadataAnnotation):
+    cte = TreeMetadataCTE
+    cte_columns = ('lft', 'rght', 'pk')
+
+    def __init__(self, *args, **kwargs):
+        self.include_self = kwargs.pop('include_self', False)
+        super(AncestorAnnotation, self).__init__(*args, **kwargs)
+
+    def build_ancestor_condition(self, cte):
+        """
+        @see MPTTModel.get_ancestors()
+        """
+        left_op = '<='
+        right_op = '>='
+
+        if not self.include_self:
+            left_op = '<'
+            right_op = '>'
+
+        return [
+            BooleanComparison(cte.col.lft, left_op, F('lft')),
+            BooleanComparison(cte.col.rght, right_op, F('rght')),
+        ]
+
+
+class DistinctJSONBAgg(JSONBAgg):
+    # DISTINCT kwarg isn't working on JSONBAgg, so enforce here
+    template = "%(function)s(DISTINCT %(expressions)s)"
+
+
+class AncestorSubquery(AncestorAnnotation):
```

Comment:
There's a potential naming clash here since Django has the notion of a `Subquery`. I wouldn't say this aligns to a Django `Subquery` by convention, but more with an `Annotation` since the class represents just a portion of a SQL `SELECT`. I'm totally fine with this as is though, just wanted to mention it.

Reply:
Good point! Renamed to AncestorArrayAgg